lib/node_modules/@stdlib/math/base/special/asinf/test/test.js

```diff
@@ -0,0 +1,151 @@
+var abs = require( '@stdlib/math/base/special/abs' );
+var float64ToFloat32 = require( '@stdlib/number/float64/base/to-float32' );
+var asinf = require( './../lib' );
+
+
+// FIXTURES //
+
+var data = require( './fixtures/julia/data.json' );
+var smallNegative = require( './fixtures/julia/small_negative.json' );
+var smallPositive = require( './fixtures/julia/small_positive.json' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof asinf, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function computes the arcsine', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var y;
+	var i;
+	var e;
+
+	x = data.x;
+	expected = data.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		y = asinf( ( ( x[ i ] ) ) );
+		e = float64ToFloat32( expected[ i ] );
+		if ( y === e ) {
+			t.equal( y, e, 'x: '+x[ i ]+'. E: '+e );
+		} else {
+			delta = abs( y - e );
+			tol = 3.3 * EPS * abs( e );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[ i ]+'. y: '+y+'. E: '+e+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function computes the arcsine (small negative values)', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var y;
+	var i;
+	var e;
+
+	x = smallNegative.x;
+	expected = smallNegative.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		y = asinf( ( x[ i ] ) );
+		e = float64ToFloat32( expected[ i ] );
+		if ( y === e ) {
+			t.equal( y, e, 'x: '+x[ i ]+'. E: '+e );
+		} else {
+			delta = abs( y - e );
+			tol = EPS * abs( e );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[ i ]+'. y: '+y+'. E: '+e+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function computes the arcsine (small positive values)', function test( t ) {
+	var expected;
+	var delta;
+	var tol;
+	var x;
+	var y;
+	var i;
+	var e;
+
+	x = smallPositive.x;
+	expected = smallPositive.expected;
+
+	for ( i = 0; i < x.length; i++ ) {
+		y = asinf( x[ i ] );
+		e = float64ToFloat32( expected[ i ] );
+		if ( y === e ) {
+			t.equal( y, e, 'x: '+x[ i ]+'. E: '+e );
+		} else {
+			delta = abs( y - e );
+			tol = EPS * abs( e );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[ i ]+'. y: '+y+'. E: '+e+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
+	}
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `NaN`', function test( t ) {
+	var v = asinf( NaN );
+	t.equal( isnanf( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided a value less than `-1`', function test( t ) {
+	var v;
+	var i;
+
+	for ( i = 0; i < 1e3; i++ ) {
+		v = uniform( -1.0e6, -1.0-EPS );
+		t.equal( isnanf( asinf( v ) ), true, 'returns NaN when provided '+v );
+	}
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided a value greater than `+1`', function test( t ) {
+	var v;
+	var i;
+
+	for ( i = 0; i < 1e3; i++ ) {
+		v = uniform( 1.0e6, 1.0 + EPS );
```

Comment:
```suggestion
		v = uniform( 1.0+EPS, 1.0e6 );
```

`a` must be less than `b`.